tmva/sofie/src/RModel.cxx

```diff
@@ -339,6 +339,10 @@ namespace SOFIE{
             fGC += "std::vector<int64_t> fTensor_" + i.first  + " = std::vector<int64_t>(" + std::to_string(length) + ");\n";
             fGC += "int64_t * tensor_" + i.first + " = fTensor_" + i.first  + ".data();\n";
          }
+         if (i.second.type == ETensorType::BOOL){
+            fGC += "bool fTensor_" + i.first  + " [" + std::to_string(length) + "] = {false};\n";
+            fGC += "bool * tensor_" + i.first + " = fTensor_"+ i.first + ";\n";
```

Comment:
Can we only define the boolean array here and then access its elements? I believe arrays in C++ reference the first element, which can be then moved if required.

So, just this should work? Otherwise, here the pointer address is getting copied to another location unnecessarily.

```
 fGC += "bool tensor_" + i.first  + " [" + std::to_string(length) + "] = {false};\n";
```